### ISSUE TITLE
회원가입 오류코드 중복

### ISSUE THREAD

**Issue by gaon12** [gaon12/eureka#11]
비밀번호와 비밀번호 확인 불일치
'''json
{
  "status": 400,
  "message": "비밀번호 재확인 필요"
}
'''
필수항목 모두 입력되지 않음
'''json
{
  "status": 400,
  "message": "필수 항목 입력 필요"
}
'''
README.md에서 두 오류의 오류코드가 동일하게 적혀져 있는데, 코드상에서도 둘 다 400 에러로 뜨는 것인지 README.md에서만 그런 것인지 궁금합니다. 로그인 오류 메세지를 보고 추측하자면 필수 항목이 모두 입력되지 않으면 401 에러가 나와야 하지 않을까 싶습니다.

**Comment by vwaiolet** [comment]
400 : 잘못된 요청
401 : 권한 없음

따라서 둘다 400으로 분류했습니다

**Comment by gaon12** [comment]
> 400 : 잘못된 요청
> 
> 401 : 권한 없음
> 
> 
> 
> 따라서 둘다 400으로 분류했습니다

헷갈릴거 같은데 혹시 둘의 코드를 분리할 수 있을까요?

**Comment by vwaiolet** [comment]
어떤 에러가 발생했는지 메시지도 함께 전달하는데 어떤 부분에서 헷갈리는지 알 수 있을까요?

**Comment by gaon12** [comment]
> 어떤 에러가 발생했는지 메시지도 함께 전달하는데 어떤 부분에서 헷갈리는지 알 수 있을까요?

나중에 통계 페이지 만들고 어떤 오류가 많이 발생했는지 확인할 때, 오류 코드를 기준으로 만들려고 합니다. 그렇기 때문에 오류 코드를 가급적이면 분리하면 좋을 것 같습니다.

추가로 오류 발생 시 db에 오류 코드와 오류 발생 날짜/시간 등도 같이 저장하면 통계 페이지 만들 때 도움이 될 것 같습니다.

**Comment by gaon12** [comment]
> 고려는 해보겠습니다
> N사에서 제공하는것처럼 http 상태코드와는 별개로 에러코드를 따로 제공하는것은 어떤가요?
> 
> 추가로 건의 주신 내용은 지금 진행하면 다른 기능 개발이 딜레이 될 수 있기에 다른 기능 모두 완성 후에 추가를 고려해보도록 하겠습니다

네! 좋습니다! 급한건 아니기 때문에 나중에 해도 괜찮습니다. 감사합니다! 일단은 이슈를 계속 오픈해 놓겠습니다!
